### PR TITLE
Enhance `Tranformations` docs

### PR DESCRIPTION
**Pull request by jofrevalles**
### Summary
This PR enhances the documentation of the `transform` functions. We fixed the previous text and, for the transformations that enabled it, we added a small visual example showing how that particular transformation work.

Moreover, we fixed a minor problem in the `DiagonalReduction` transformation, where the COPY tensor created did not have the `dual` metadata, which is necessary to display the transformed tensor network correctly.

### CODE REVIEW

File: docs/src/transformations.md [hunk 1]
```diff
@@ -14,41 +14,300 @@ transform!
...
+```@example plot
 using QuacIO
-using CairoMakie
-using Tenet
+set_theme!(resolution=(800,400)) # hide
 
 sites = [5, 6, 14, 15, 16, 17, 24, 25, 26, 27, 28, 32, 33, 34, 35, 36, 37, 38, 39, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 61, 62, 63, 64, 65, 66, 67, 72, 73, 74, 75, 76, 83, 84, 85, 94]
 circuit = QuacIO.parse(joinpath(@__DIR__, "sycamore_53_10_0.qasm"), format=QuacIO.Qflex(), sites=sites)
 tn = TensorNetwork(circuit)
 transformed_tn = transform(tn, Tenet.RankSimplification)
```
Comment (mofeing) [inline docs/src/transformations.md:65]: Would you mind adding some more transformations here? I know that quimb's `full_simplify` function uses `AntiDiagonalGauging`, `DiagonalReduction`, `ColumnReduction` and `RankSimplification` in that order by default.

**Review by mofeing — CHANGES_REQUESTED** [review]
It looks AWESOME!
Just some lil requests.

File: docs/src/transformations.md (hunk 1 above)
Comment (jofrevalles) [reply docs/src/transformations.md:65]: Sure!

**Comment by mofeing** [comment]
I think it would look better if we shorten the caption below the figures. Something like "Reference" or "Original" and "Transformed" for example. What do you think?

**Comment by jofrevalles** [comment]
> I think it would look better if we shorten the caption below the figures. Something like "Reference" or "Original" and "Transformed" for example. What do you think?

Yes, I think that is a good idea! I will do that now.

File: docs/src/transformations.md [hunk 2]
```diff
@@ -14,41 +14,300 @@ transform!
 ```
 
 # Example
```
Comment (mofeing) [inline docs/src/transformations.md:16]:
```suggestion
# Example: RQC simplification
```

File: docs/src/transformations.md (hunk 1 above)
Comment (jofrevalles) [reply docs/src/transformations.md]: I have done that but it appears that for this specific case we obtain the same network that we would obtain with only `RankSimplification`. Nevertheless, we would expect some advantage for some larger circuits when we apply all the transformations.

File: src/Transformations.jl [hunk 3]
```diff
@@ -345,6 +345,10 @@ function find_zero_columns(x; atol = 1e-12)
 end
 
 function find_diag_axes(x; atol = 1e-12)
+    if ndims(parent(x)) == 1 # Skip 1D tensors
+        return []
+    end
```
Comment (mofeing) [inline src/Transformations.jl:351]:
```suggestion
    # skip 1D tensors
    ndims(parent(x)) == 1 && return []
```

File: src/Transformations.jl [hunk 4]
```diff
@@ -367,6 +371,10 @@ function find_diag_axes(x; atol = 1e-12)
 end
 
 function find_anti_diag_axes(x; atol = 1e-12)
+    if ndims(parent(x)) == 1 # Skip 1D tensors
+        return []
+    end
```
Comment (mofeing) [inline src/Transformations.jl:377]:
```suggestion
    # skip 1D tensors
    ndims(parent(x)) == 1 && return []
```

**Review by mofeing — CHANGES_REQUESTED** [review]
Just some aesthetic requests.

File: docs/src/transformations.md [hunk 5]
```diff
@@ -13,42 +13,301 @@ transform
...
+Label(fig[1, 2, Bottom()], "Transformed") #hide
+
+fig #hide
+```
+
+```@docs
 Tenet.DiagonalReduction
-Tenet.RankSimplification
+```
+
```
Comment (mofeing) [inline docs/src/transformations.md:151]:
```suggestion
```@example plot
set_theme!(resolution=(800,200)) # hide
fig = Figure() #hide

data = zeros(Float64, 2, 2, 2, 2) #hide
for i in 1:2 #hide
    for j in 1:2 #hide
        for k in 1:2 #hide
            data[i, i, j, k] = k #hide
        end #hide
    end #hide
end #hide

A = Tensor(data, (:i, :j, :k, :l)) #hide
B = Tensor(rand(2, 2), (:i, :m)) #hide
C = Tensor(rand(2, 2), (:j, :n)) #hide

tn = TensorNetwork([A, B, C]) #hide
reduced = transform(tn, Tenet.DiagonalReduction) #hide

smooth_annotation!( #hide
    fig[1, 1]; #hide
    color = bg_blue, #hide
    xlims = [-2, 2], #hide
    ylims = [-2, 2], #hide
    offset_x = -0.21, #hide
    offset_y = -0.42, #hide
    radius_x = 0.38, #hide
    radius_y = 0.8, #hide
    num_waves = 6, #hide
    fluctuation_amplitude = 0.02, #hide
    phase_shift = 0.0) #hide
plot!(fig[1, 1], tn, layout=Spring(iterations=1000, C=0.5, seed=100); node_color=[red, orange, orange, :black, :black,:black, :black]) #hide

smooth_annotation!( #hide
    fig[1, 2]; #hide
    color = bg_blue, #hide
    xlims = [-2, 2], #hide
    ylims = [-2, 2], #hide
    offset_x = 0.1, #hide
    offset_y = -0.35, #hide
    radius_x = 0.38, #hide
    radius_y = 1.1, #hide
    num_waves = 5, #hide
    fluctuation_amplitude = 0.02, #hide
    phase_shift = 1.9) #hide
plot!(fig[1, 2], reduced, layout=Spring(iterations=1000, C=0.5, seed=100),  node_color=[orange, orange, red, :black, :black, :black, :black, :black]) #hide

Label(fig[1, 1, Bottom()], "Original") #hide
Label(fig[1, 2, Bottom()], "Transformed") #hide

fig #hide
```

**Pull request merged**